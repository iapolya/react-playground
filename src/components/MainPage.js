import React, { useEffect } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import '../containers/App.css';

import { Header } from './Header';

const MainPage = (props) => {
  const { searchField, onSearchChange, cats, isPending } = props;

  useEffect(() => {
    props.onRequestCats();
  }, []);

  let filteredCats = (cats) =>
    cats.filter((cat) => {
      return cat.name.toLowerCase().includes(searchField.toLowerCase());
    });

  return (
    <div className="tc">
      <Header />
      {isPending ? (
        <div>loading</div>
      ) : (
        <Scroll>
          <SearchBox searchField={searchField} searchChange={onSearchChange} />
          <CardList cats={filteredCats(cats)} />
        </Scroll>
      )}
    </div>
  );
};

export default MainPage;
