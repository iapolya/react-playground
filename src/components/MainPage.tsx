import React, { useEffect, ChangeEventHandler } from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import '../containers/App.css';

import { Header } from './Header';
import { ICat, ISearch } from "../types";

export interface IMainProps extends ISearch {
  cats: Array<ICat>,
  isPending: Boolean,
  onRequestCats: Function
}

const MainPage = (props: IMainProps) => {
  const { searchField, onSearchChange, cats, isPending } = props;

  useEffect(() => {
    props.onRequestCats();
  }, []);

  let filteredCats = (cats: Array<ICat>) =>
    cats.filter((cat: ICat) => {
      return cat.name.toLowerCase().includes(searchField.toLowerCase());
    });

  return (
    <div className="tc">
      <Header />
      {isPending ? (
        <div>loading</div>
      ) : (
        <Scroll>
          <SearchBox searchField={searchField} searchChange={onSearchChange} />
          <CardList cats={filteredCats(cats)} />
        </Scroll>
      )}
    </div>
  );
};

export default MainPage;
