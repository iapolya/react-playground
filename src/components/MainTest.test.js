import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';
// redux-mock-store can be used

let wrapper;
beforeEach(() => {
  const mockProps = { searchField: '', onSearchChange: jest.fn(), cats: [], isPending: false };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it('expect to render app', () => {
  expect(wrapper).toMatchSnapshot();
});
