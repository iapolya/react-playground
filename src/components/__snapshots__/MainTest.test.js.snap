// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render app 1`] = `
<div
  className="tc"
>
  <Header />
  <Scroll>
    <SearchBox
      searchChange={[MockFunction]}
      searchField=""
    />
    <CardList
      cats={Array []}
    />
  </Scroll>
</div>
`;
